### ISSUE TITLE
[APP SUBMITTED]: SEARCHQUEUE-BACKLOG-75897 :: [HDBits] :: rTorrent: Unable to send Torrent: Return code undefined

### ISSUE THREAD

**Issue by GoodOmens83** [SickRage/sickrage-issues#67]
### INFO
Python Version: **2.7.10 (default, Jun  4 2015, 01:18:03) [GCC 4.3.2]**
Operating System: **Linux-2.6.32.12-ppc-with-glibc2.0**
Locale: unknown
Branch: **master**
Commit: SickRage/SickRage@91e54c0364df98358a01436ee247a348775e88e1
Link to Log: https://gist.github.com/b268b2143a837703589b
### ERROR
```
SEARCHQUEUE-BACKLOG-75897 :: [HDBits] :: rTorrent: Unable to send Torrent: Return code undefined
```
---
_STAFF NOTIFIED_: @SickRage/owners @SickRage/moderators

**Comment by fernandog** [comment]
Known issue.
We need to "read" the client response (JSON) and not the URL return code.

**Comment by labrys** [comment]
@fernandog is this fix still pending?

**Comment by fernandog** [comment]
Yes. I did a PR (not approved) in the old repo to fix (possible) this.
Its related to the TODO:

Check clients return message (JSON) and not only return_code

I think @miigotu have another idea to fix this

**Comment by labrys** [comment]
Ok I'm going to remove the fix pending tag then since we don't currently have a fix in the works.

**Comment by fernandog** [comment]
@GoodOmens83 we need to improve the hdbits code, do you have an invite so @miigotu can work on it?

**Comment by GoodOmens83** [comment]
@fernandog Unfortunately not as I'm seeding under the required 2TB for invites.  I reached out to the staff to see if they can accommodate but I'm assuming not.  I'll see what else I can do.

**Comment by fernandog** [comment]
@GoodOmens83 
maybe you can give us the HTML code doing a search for "game of thrones s05" in all tv categories

**Comment by GoodOmens83** [comment]
@fernandog I actually just reached out to HDBits and they are willing to send a invite.  I emailed @miigotu

**Comment by GoodOmens83** [comment]
@miigotu Invite has been sent.  @fernandog  HDBit's JSON API is documented in the Wiki on their site.  Want me to send you some screenshots of it?

**Comment by miigotu** [comment]
nope that's OK, I'll check their wiki. thanks

**Comment by miigotu** [comment]
@GoodOmens83 wrong hdbits. Our provider is for hdbits.org, not hd-bits.com

**Comment by GoodOmens83** [comment]
That's precisely the site I had an invite sent from and you created an
account on.

**Comment by miigotu** [comment]
now I feel stupid lol, I got sent 2 invites near the same time and didnt realize I had both

**Comment by fernandog** [comment]
looool

@miigotu we should rename our HDBITS to HDBITS.ORG

**Comment by GoodOmens83** [comment]
So an update:

Error still persists but now it's declaring because rTorrent has timed out (I'm assuming because of proper JSON error code reading now).  

Still only happens for backlog searches.  I've looked through the code and can't see why the call to rTorrent would be different for a backlog search vs. a normal daily search.

e.g.,
2016-03-01 19:22:21 INFO     SEARCHQUEUE-BACKLOG-121361 :: [HDBits] :: Every ep in this season is needed, downloading the whole torrent 
2016-03-01 19:22:21 INFO     SEARCHQUEUE-BACKLOG-121361 :: Downloading xyz from HDBits
2016-03-01 19:22:22 DEBUG    SEARCHQUEUE-BACKLOG-121361 :: GET URL: xyz[Status: 200]
2016-03-01 19:22:25 DEBUG    SEARCHQUEUE-BACKLOG-121361 :: Calling rTorrent Client
2016-03-01 19:22:55 WARNING  SEARCHQUEUE-BACKLOG-121361 :: Error while sending torrent: timed out [ace5e56]
2016-03-01 19:22:55 WARNING  SEARCHQUEUE-BACKLOG-121361 :: rTorrent: Unable to send Torrent [ace5e56]
